code/modules/food_and_drinks/food.dm

```diff
@@ -32,13 +36,13 @@
 					to_chat(H,"<span class='warning'>What the hell was that thing?!</span>")
 					H.adjust_disgust(25 + 30 * fraction)
 					SEND_SIGNAL(H, COMSIG_ADD_MOOD_EVENT, "toxic_food", /datum/mood_event/disgusting_food)
-				else if(foodtype & H.dna.species.disliked_food)
+				else if((foodtype & H.dna.species.disliked_food) || food_quality <= 30)
 					to_chat(H,"<span class='notice'>That didn't taste very good...</span>")
 					H.adjust_disgust(11 + 15 * fraction)
 					SEND_SIGNAL(H, COMSIG_ADD_MOOD_EVENT, "gross_food", /datum/mood_event/gross_food)
-				else if(foodtype & H.dna.species.liked_food)
+				else if((foodtype & H.dna.species.liked_food & food_quality >= 50) || food_quality >= 70) //you like food of high quality, and food of regular quality you have a preference for
```

Comment:
```suggestion
				else if(((foodtype & H.dna.species.liked_food) && food_quality >= 50) || food_quality >= 70) //you like food of high quality, and food of regular quality you have a preference for
```
you put a bitwise operator instead of a boolean